src/common/feedGenerator.ts

```diff
@@ -77,10 +79,26 @@ export const feedToFilters = async (
             'adv.id = fas."advancedSettingsId" AND fas."feedId" = :feedId',
             { feedId },
           )
-          .where('COALESCE(fas.enabled, adv.defaultEnabledState) = false')
-          .getQuery();
+          .where('COALESCE(fas.enabled, adv.defaultEnabledState) = false');
+
+        const experimentSettings = features?.advanced_settings_default_values;
+        if (!experimentSettings?.enabled) {
+          return `s.advancedSettings && array(${subQuery.getQuery()})`;
+        }
+
+        const settings = Object.keys(experimentSettings.value).map((key) => ({
+          id: parseInt(key),
+          defaultEnabledState: experimentSettings.value[key],
+        }));
+
+        settings.forEach((experiment) => {
+          subQuery = subQuery.orWhere(
+            `(adv.id = :id AND COALESCE(fas.enabled, :enabled, adv.defaultEnabledState) = false)`,
+            { id: experiment.id, enabled: experiment.defaultEnabledState },
+          );
+        });
```

Comment:
This can be refactored into a single reduce instead of a map and forEach.
The reduce will iterate over the keys of the experiments settings and will return the updated sub query